### PR TITLE
Remove default timeout from RequestUtil

### PR DESCRIPTION
**Pull request by jfurrow**
This PR changes `RequestUtil#json` in the following ways...
- It will not send a request if a previous request to the same URL has not yet resolved
  - Before, it did this only if `hangingRequestCallback` was defined
- No timeout is applied by default
  - Before, it applied a 2 second timeout if `hangingRequestCallback` was not defined
- Constructs an ID for each request based on the request's `contentType`, `data`, `headers`, `method`, and `url`.

Previous functionality (some requests fail after 2 seconds):

New functionality (requests wait indefinitely):

### CODE REVIEW

File: utils/RequestUtil.js [hunk 1]
```diff
@@ -73,20 +73,23 @@ var RequestUtil = {
   json: function (options) {
     // Default assign options to empty object
     options || (options = {});
+    var requestID = options.url;
```
Comment (rcorral) [inline utils/RequestUtil.js]: This isn't part of this PR, but we should actually create a better unique id. Sometimes there's different query parameters that we pass or headers. Could you think of a good way to create a unique id out of all of this?
Also, keep in mind that `_timestamp` needs to be ommitted from the params.

Comment (jfurrow) [inline utils/RequestUtil.js]: So does this mean that each time the query params are different, that should result in a unique ID? Or should query params not affect the ID?

Comment (MatApple) [inline utils/RequestUtil.js]: Query params should effect the ID. So should the method, headers and possibly data for POST, PUT, PATCH.

Comment (rcorral) [inline utils/RequestUtil.js]: @jfurrow how do you feel about making this change in this PR?

Comment (jfurrow) [inline utils/RequestUtil.js]: Sure! Will start on this in just a moment.

Comment (jfurrow) [inline utils/RequestUtil.js]: @rcorral @MatApple I just pushed a commit which constructs the ID based on the properties `contentType`, `data`, `headers`, `method`, `url`. Let me know if you think this is sufficient!

**Comment by rcorral** [comment]
Looks great, thanks John!

**Pull request merged**